src/bridging/providers/bungee/BungeeBridgeProvider.ts

```diff
@@ -0,0 +1,232 @@
+import { Signer } from 'ethers'
+import { latest as latestAppData } from '@cowprotocol/app-data'
+
+import {
+  BridgeDeposit,
+  BridgeHook,
+  BridgeProvider,
+  BridgeProviderInfo,
+  BridgeQuoteResult,
+  BridgeStatusResult,
+  QuoteBridgeRequest,
+} from '../../types'
+
+import { DEFAULT_GAS_COST_FOR_HOOK_ESTIMATION, RAW_PROVIDERS_FILES_PATH } from '../../const'
+
+import { ChainId, ChainInfo, SupportedChainId, TargetChainId } from '../../../chains'
+
+import { BUNGEE_TOKEN_MAPPING } from './const/tokens'
+import { EvmCall, TokenInfo } from '../../../common'
+
+import { mainnet } from '../../../chains/details/mainnet'
+import { polygon } from '../../../chains/details/polygon'
+import { arbitrumOne } from '../../../chains/details/arbitrum'
+import { base } from '../../../chains/details/base'
+import { optimism } from '../../../chains/details/optimism'
+import { BungeeApi, BungeeApiOptions } from './BungeeApi'
+import { getChainConfigs, getTokenAddress, getTokenSymbol, toBridgeQuoteResult } from './util'
+import { CowShedSdk, CowShedSdkOptions } from '../../../cow-shed'
+import { createBungeeDepositCall } from './createBungeeDepositCall'
+import { OrderKind } from '@cowprotocol/contracts'
+import { HOOK_DAPP_BRIDGE_PROVIDER_PREFIX } from './const/misc'
+import { BungeeBuildTx, BungeeQuote, BungeeQuoteAPIRequest } from './types'
+import { getSigner } from 'src/common/utils/wallet'
+
+const HOOK_DAPP_ID = `${HOOK_DAPP_BRIDGE_PROVIDER_PREFIX}/bungee`
+export const BUNGEE_SUPPORTED_NETWORKS = [mainnet, polygon, arbitrumOne, base, optimism]
+
+/** There will be no dex swaps happening while bridging. Hence slippage will be zero */
+const SLIPPAGE_TOLERANCE_BPS = 0
+
+export interface BungeeBridgeProviderOptions {
+  /**
+   * Token info provider
+   * @param chainId - The chain ID
+   * @param addresses - The addresses of the tokens to get the info for
+   * @returns The token infos
+   */
+  getTokenInfos?: (chainId: ChainId, addresses: string[]) => Promise<TokenInfo[]>
+
+  // API options
+  apiOptions?: BungeeApiOptions
+
+  // Cow-shed options
+  cowShedOptions?: CowShedSdkOptions
+}
+
+export interface BungeeQuoteResult extends BridgeQuoteResult {
+  bungeeQuote: BungeeQuote
+  buildTx: BungeeBuildTx
+}
+
+export class BungeeBridgeProvider implements BridgeProvider<BungeeQuoteResult> {
+  protected api: BungeeApi
+  protected cowShedSdk: CowShedSdk
+
+  constructor(private options: BungeeBridgeProviderOptions = {}) {
+    this.api = new BungeeApi(options.apiOptions)
+    this.cowShedSdk = new CowShedSdk(options.cowShedOptions)
+  }
+
+  info: BridgeProviderInfo = {
+    name: 'Bungee',
+    logoUrl: `${RAW_PROVIDERS_FILES_PATH}/bungee/bungee-logo.png`,
+  }
+
+  async getNetworks(): Promise<ChainInfo[]> {
+    return BUNGEE_SUPPORTED_NETWORKS
+  }
+
+  async getBuyTokens(targetChainId: TargetChainId): Promise<TokenInfo[]> {
```

Comment:
If you have an API which return list of supported tokens, would be great to add it here. An example: https://github.com/cowprotocol/cow-sdk/pull/322